### ISSUE TITLE
Mcedit 1.5.5.0 Not updating command Block Coordinates Upon Paste

### ISSUE THREAD

**Issue by Julienc249** [Khroki/MCEdit-Unified#762]
Every time I paste a schematic containing command blocks with static coordinates the Static coordinates do not get updated in ANY of the Command Blocks. This is started happening in the newest version (1.5.5.0). The command do retain their data, simply they do not get updated. Thank-you for any solutions that can be offered.

**Comment by LaChal** [comment]
Please, answer of the following question, even if some may look stupid:

* What is the game version of the world you want to import schematics in?
* If you know it, what is the game version used to create the schematics?
* In the import tool panel, did you check the box `Update Command Block Coords`?

Also, note that, if the option `Cancel Command Block Offset` is enabled, this will prevent all coordinates in command blocks to be updated. You can find the related checkbox in the main `options` panel.

**Comment by Julienc249** [comment]
Hello, the world is on minecraft game version: 1.11.2 for both worlds (paste and schematic)
I did make sure to check the box - Update Command Block Coords. 
I will Try with that last note you gave me! Thank-you!

**Comment by LaChal** [comment]
We've changed the game IDs handling in MCEdit-Unified v1.5.5.0 to make it more friendly with different game versions. It is still a work in progress feature, and will be enhanced in future versions.

Having the files needed by the program to handle correctly the game v1.11.2 may solve your issue.

To get them, just copy the `1.11` folder as `1.11.2` in the `mcver` one you'll find in your MCEdit installation directory.
If you run on Mac, you'll have to enter the .app.
If you run on Linux, creating a link to the `1.11` folder shall do the job.

**Comment by Khroki** [comment]
In the future this feature may be enhanced to automatically fall back to the next lowest version to prevent this (1.9.2  using the 1.9 defs etc.) I think some improvements were made for 1.5.6.0 if I remember correctly though.

**Comment by LaChal** [comment]
Right.
The files can now "call" other ones when loading. The MCEdit v1.5.6.0 files for game v1.11.2 contain one definition each to make the 1.11 files be loaded.
An automatic fall back to the next lowest version will be also added.
Note that PE also has its own directory and files :smile:

**Comment by Julienc249** [comment]
Hello, I have been unable to find the mcver folder and the new update has not fixed the problem. When I do not rotate the ship the commands remain within the command blocks but are un-changed (even with the command update option on) When I rotate the schematic the commands are all wiped. I would really appreciate any suggestion on how to fix these specific problems. 
Thank-you

**Comment by LaChal** [comment]
The `mcver` folder shall be included in the installers...
We'll have a look at this.

You can try the following steps:

1. Open this link: https://github.com/Khroki/MCEdit-Unified/tree/4246e24d58d8cfb5b83f6328362b3ee674e46686  
It is the code which was used to build v1.5.6.0.
2. Click on the `Clone or Download` button, then on `Download ZIP`.
3. Save or open the archive. You'll have the `mcver` folder in.
4. Put the `mcver` in the folder where MCEdit is installed.  
If you run on Mac, you'll have to open the .app.  
If you run on Windows, put the folder alongside `mcedit.exe`.  
If you run on Linux, put the folder alongside `mcedit.sh`.

**Comment by Julienc249** [comment]
I did exactly that and yet the problem remains the same. I run on windows - Here is a screen shot of where the folder is placed: 
![screenshot 9](https://cloud.githubusercontent.com/assets/25248240/22398374/51b990aa-e56e-11e6-9dc7-f0363c26037d.png)

**Comment by LaChal** [comment]
@Julienc249 Did you try MCEdit v1.5.6.0?

**Comment by Julienc249** [comment]
I am sorry to have to reopen this thread again, however, lately, every time I rotate and paste a schematic all nbt data is wiped (command blocks are wiped, chests, paintings, signs go invisible etc.) I cannot understand why this is happening. Could it have to do with the fact that the world the schematics were taken from is from 1.8 and the world I am pasting them in is 1.9 - 1.11.2? 

Thank-you for any help you can provide and fix you can come up with.

**Comment by LaChal** [comment]
I don't experience this on my side.

Which MCEdit version are you using?

**Comment by Julienc249** [comment]
Sorry for the late response, I am using the latest version of mcedit unified - 1.5.6.0 I believe.

**Comment by LaChal** [comment]
To know the exact version of MCEdit, just look in the console, or the title bar :)

On my side, as I said, I don't experience this when importing MC 1.8 schematics I created.
Can you send your's, please?
We'll have a look at, and try to find the cause.

**Comment by Julienc249** [comment]
Ok, thank-you very much, it is indeed the latest version (1.5.6.0) and the main schematic that seems to exhibit the problem is the one that you should receive through gmail. (hopefully if this one's cause can be tracked down, my other can be fixed as well) 

Again, thank-you very much!

**Comment by LaChal** [comment]
No incoming stuff from gmail for now...

I guess you can upload it here, though.

**Comment by Julienc249** [comment]
[U.S.S. Valiant.zip](https://github.com/Khroki/MCEdit-Unified/files/1122986/U.S.S.Valiant.zip)

**Comment by Julienc249** [comment]
After doing some testing, I realized that maybe what is happening is that the command blocks are not getting rotated but actually replaced, with empty nbt data. Maybe this could be due to the 1.9 updates where command block rotation and facing began to have an impact on their functionality.

**Comment by Julienc249** [comment]
Hello, does anyone know how this problem is going?

**Comment by Julienc249** [comment]
To anyone that is interested: I believe now that this issue is directly linked to the issue with paintings and item frames issue that has been observed before. This issue is only occurring in the latest version of mcedit. One version lower and you can avoid the problem. However, all paintings are cleared and command blocks are not updated despite the update command block coords being selected. We can only hope that this will be fixed soon.

**Comment by LaChal** [comment]
@Julienc249 I've made some tests on my side with the 'U.S.S. Valiant' schematic you provided.
The fact is some command blocks do not have any command in the schematic. 5I guess you already knew that.)
Also, the NBT data looks good for paintings and item frames and are imported by MCEdit, but once loaded in MC 1.12-pre7, they are removed...
When importing the schematic in a 1.10.2 world, the signs are still there, but no visible.

From which game version the schematic has been created?

**Comment by Julienc249** [comment]
Yes, many command blocks are empty and that is normal, however all of them get cleared (you already know that) the schematic was taken in 1.8.8, just before the release of 1.9. Thank-you very much for the work you have done already tracking this issue down.
